dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -129,4 +138,25 @@ public void createAddTagToEntryEvent(User user, Entry entry, Version version) {
             create(event);
         }
     }
+
+    public void publishEvent(boolean publish, User user, Entry entry) {
+        Class clazz = entry.getClass();
```

Comment:
You probably tried these, but why not `instanceof` or `Entry.getEntryType()` or `Entry.getEventBuilder` instead of reflection?

Reply:
Yikes, I was not aware of `Entry.getEventBuilder` -- that's much better!

Reply:
Fixed in 15336f0a1ec56182d1b867fe5c01feb54d632f2c